h2o-core/src/main/java/water/AutoBuffer.java

```diff
@@ -1475,7 +1475,8 @@ public static Object javaSerializeReadPojo(byte [] bytes) {
     try {
       return new ObjectInputStream(new ByteArrayInputStream(bytes)).readObject();
     } catch (IOException e) {
-      throw Log.throwErr(e);
+      String className = new String(bytes, 8, Math.min(40, Math.max(3, bytes[7])));
```

Comment:
Could this possibly lead to ArrayIndexOutOfBounds exception? This method doesn't make any assumptions on the length of the byte array.

It would be great if this one liner could be turned into a helper function (not sure exactly where it should go). It can also have a unit test.